lib/memoize_block.rb

```diff
@@ -23,9 +23,9 @@ module MemoizeBlock
   # end
   #
   def memoize(ivar_name = nil)
-    ivar_name = caller_locations[0].label if ivar_name.nil?
+    ivar_name = caller_locations[0].label.gsub('?', '') if ivar_name.nil?
```

Comment:
Does this open us up to potential collisions? For example:

```ruby
def thing
  memoize { foo }
end

def thing?
  memoize { true unless foo.nil? }
end
```

Reply:
Good point, that's actually technically true now with it if you were to write out a custom instance variable name as well, but this is probably worth handling since this is dynamic. 

How about a new schema to map a `?`? Maybe double underscore or something with low risk of collision `__`

Reply:
Yeah, that sounds like a good plan. I don't think so names really ever surface aside from possibly in a stacktrace so anything is likely fine--the double underscore like you have above, `_boolean`, `_question_mark`, etc.